### PR TITLE
241 SIM gh-action-to-run-unit-tests-on-pr-open-and-approvals

### PR DESCRIPTION
**Pull request by AgustinRamiroDiaz**
# What

This only runs the frontend unit tests

# Testing done

The tests have triggered on opening and reopening the PR

# Pending for future iterations

Out of scope:
- Backend unit tests
- [Trigger from comments](https://genlayer.slack.com/archives/C071DE6200M/p1719851490588059)
  - I'm still researching this approach to see if it's possible given that [some comments](https://github.com/orgs/community/discussions/25389#discussioncomment-8935283) mention it doesn't behave like PR checks

### CODE REVIEW

File: .github/workflows/frontend-unit-tests.yml
Comment (AgustinRamiroDiaz) [inline .github/workflows/frontend-unit-tests.yml:1]: Decoupling workflows is useful for modular approach, they work as if they were functions that can be called

File: .github/workflows/unit-tests-pr.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - ready_for_review
+      - reopened
+  pull_request_review:
+    types:
+      - submitted
+      - edited
+
+concurrency:
+  group: unit-tests-${{ github.event.number }}
+  cancel-in-progress: true
```
Comment (AgustinRamiroDiaz) [inline .github/workflows/unit-tests-pr.yml:16]: This is useful to discard older jobs on new triggers

File: .github/workflows/unit-tests-pr.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: Unit Tests
+    if: ${{ github.event.review.state == 'APPROVED' || github.event_name == 'pull_request' }}
```
Comment (AgustinRamiroDiaz) [inline .github/workflows/unit-tests-pr.yml:24]: this is the way to check if the review was an approve as per https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved

**Comment by cristiam86** [comment]
@AgustinRamiroDiaz could you please change the title of the PR to have the issue number as the other PRs?

**Review by cristiam86 — APPROVED** [review]

**Pull request merged**